### PR TITLE
Fixed model generation with same names

### PR DESCRIPTION
**Pull request by drpayyne**
**Description**

Fixed generation of models with same model/resource model name.

**Fixed Issues**

1. Fixes magento/magento2-phpstorm-plugin#437: The added model and collection classes contain duplicate names of the imported classes

**Contribution checklist**
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with integration/functional tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModelsDialog.java [hunk 1]
```diff
@@ -223,14 +223,23 @@ private void onOK() {
     private PsiFile generateModelFile() {
         final NamespaceBuilder modelNamespace = getModelNamespace();
         final NamespaceBuilder resourceModelNamespace = getResourceModelNamespace();
+        final StringBuilder resourceModelFqn
+                = new StringBuilder(resourceModelNamespace.getClassFqn());
+        String resorceModelName = getResourceModelName();
+
+        if (getModelName().equals(getResourceModelName())) {
+            resourceModelFqn.append(" as ResourceModel");
+            resorceModelName = "ResourceModel";
```
Comment (eduard13) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModelsDialog.java:232]: Typo here
```suggestion
            resourceModelName = "ResourceModel";
```

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModelsDialog.java [hunk 2]
```diff
@@ -223,14 +223,23 @@ private void onOK() {
     private PsiFile generateModelFile() {
         final NamespaceBuilder modelNamespace = getModelNamespace();
         final NamespaceBuilder resourceModelNamespace = getResourceModelNamespace();
+        final StringBuilder resourceModelFqn
+                = new StringBuilder(resourceModelNamespace.getClassFqn());
+        String resorceModelName = getResourceModelName();
+
+        if (getModelName().equals(getResourceModelName())) {
+            resourceModelFqn.append(" as ResourceModel");
```
Comment (eduard13) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModelsDialog.java:231]: Can we extract the `ResourceModel` into a const and reuse it in all places we use it?

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModelsDialog.java [hunk 3]
```diff
@@ -250,17 +259,31 @@ private PsiFile generateCollectionFile() {
         final NamespaceBuilder resourceModelNamespace = getResourceModelNamespace();
         final NamespaceBuilder modelNamespace = getModelNamespace();
         final NamespaceBuilder collectionNamespace = getCollectionNamespace();
+        final StringBuilder modelFqn = new StringBuilder(modelNamespace.getClassFqn());
+        String modelName = getModelName();
+        final StringBuilder resourceModelFqn
+                = new StringBuilder(resourceModelNamespace.getClassFqn());
+        String resorceModelName = getResourceModelName();
```
Comment (eduard13) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModelsDialog.java:266]:
```suggestion
        String resourceModelName = getResourceModelName();
```

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModelsDialog.java [hunk 4]
```diff
@@ -250,17 +259,31 @@ private PsiFile generateCollectionFile() {
...
+
+
+        if (getModelName().equals(getResourceModelName())) {
+            modelFqn.append(" as Model");
+            modelName = "Model";
+            resourceModelFqn.append(" as ResourceModel");
+            resorceModelName = "ResourceModel";
```
Comment (eduard13) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModelsDialog.java:273]:
```suggestion
            resourceModelName = "ResourceModel";
```

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModelsDialog.java [hunk 5]
```diff
@@ -250,17 +259,31 @@ private PsiFile generateCollectionFile() {
...
-            getModelName(),
+            modelName,
             getCollectionName(),
             collectionNamespace.getClassFqn(),
             getCollectionDirectory(),
             collectionNamespace.getNamespace(),
-            getResourceModelName(),
-            resourceModelNamespace.getClassFqn(),
-            modelNamespace.getClassFqn()
+            resorceModelName,
```
Comment (eduard13) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModelsDialog.java:284]:
```suggestion
            resourceModelName,
```

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModelsDialog.java [hunk 6]
```diff
@@ -223,14 +223,23 @@ private void onOK() {
     private PsiFile generateModelFile() {
         final NamespaceBuilder modelNamespace = getModelNamespace();
         final NamespaceBuilder resourceModelNamespace = getResourceModelNamespace();
+        final StringBuilder resourceModelFqn
+                = new StringBuilder(resourceModelNamespace.getClassFqn());
+        String resorceModelName = getResourceModelName();
```
Comment (eduard13) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModelsDialog.java:228]:
```suggestion
        String resourceModelName = getResourceModelName();
```

**Review by eduard13 — CHANGES_REQUESTED** [review]
Thank you for the fix, could you please check my comments below?

**Comment by drpayyne** [comment]
Hi @eduard13, thanks for catching my typos! I've implemented your suggested changes. Please review again.

**Review by eduard13 — APPROVED** [review]
Good job 👍

**Pull request merged**